SELearning.Infrastructure.Tests/Content/ContentRepositoryTests.cs

```diff
@@ -21,19 +24,389 @@ public ContentRepositoryTests()
+
+        Assert.Collection(allContent,
+            content => Assert.Equal(contentDto1, content),
+            content => Assert.Equal(contentDto2, content),
+            content => Assert.Equal(contentDto3, content),
+            content => Assert.Equal(contentDto4, content)
+        );
+    }
+
+
+    [Fact]
+    public async Task UpdateContentAsync_updates_existing_content()
+    {
+        var content = new ContentUpdateDto { Section = _section, Author = "author", Title = "title", Description = "description", VideoLink = "video link", Rating = 3 };
+
+        var updated = await _repository.UpdateContent(1, content);
+
+        Assert.Equal(OperationResult.Updated, updated);
+    }
+
+    [Fact]
+    public async Task UpdateContentAsync_given_non_existing_Content_returns_NotFound()
+    {
+        var content = new ContentUpdateDto { Section = _section, Author = "author", Title = "title", Description = "description", VideoLink = "video link", Rating = 3 };
+
+        var response = await _repository.UpdateContent(42, content);
+
+        Assert.Equal(OperationResult.NotFound, response);
+    }
+
+
+    [Fact]
+    public async Task UpdateContentAsync_updates_and_returns_Updated()
+    {
+        var contentDto = new ContentUpdateDto { Section = _section, Author = "author", Title = "new title", Description = "description", VideoLink = "VideoLink", Rating = 3 };
+
+        var response = await _repository.UpdateContent(1, contentDto);
+
+        var entity = await _context.Content.FirstAsync(c => c.Title == "new title");
+
+        Assert.Equal(OperationResult.Updated, response);
+    }
+
+    [Fact]
+    public async Task DeleteContentAsync_given_non_existing_Id_returns_NotFound()
+    {
+        var response = await _repository.DeleteContent(42);
+
+        Assert.Equal(OperationResult.NotFound, response);
+    }
+
+    [Fact]
+    public async Task DeleteContentAsync_deletes_and_returns_Deleted()
+    {
+        var response = await _repository.DeleteContent(2);
+
+        var entity = await _context.Content.FindAsync(2);
+
+        Assert.Equal(OperationResult.Deleted, response);
+        Assert.Null(entity);
+    }
+
+
+    protected virtual void Dispose(bool disposing)
+    {
+        if (!disposedValue)
+        {
+            if (disposing)
+            {
+                // TODO: dispose managed state (managed objects)
+            }
+
+            // TODO: free unmanaged resources (unmanaged objects) and override finalizer
+            // TODO: set large fields to null
+            disposedValue = true;
+        }
     }
 
+    // // TODO: override finalizer only if 'Dispose(bool disposing)' has code to free unmanaged resources
+    // ~CharacterRepositoryTests()
+    // {
+    //     // Do not change this code. Put cleanup code in 'Dispose(bool disposing)' method
+    //     Dispose(disposing: false);
+    // }
```

Comment:
What is this? Left-over junk code that hasn't been deleted, or something we should implement?